### PR TITLE
feat: grpo rollout correction for training-inference mismatch

### PR DESCRIPTION
**Pull request by hjh0119**
## Summary
This PR introduces rollout correction mechanisms to address training-inference mismatch in GRPO training, and includes several related improvements.

### 1. Support Truncated/Masked Importance Sampling (TIS/MIS) for Training-Inference Mismatch Correction
Added importance sampling correction to handle off-policy issues

New parameters:
- rollout_importance_sampling_mode: Correction mode selection
- token_truncate: Token-level truncated IS (clips extreme weights)
	- token_mask: Token-level masked IS (discards extreme tokens)
	- sequence_truncate: Sequence-level truncated IS
	- sequence_mask: Sequence-level masked IS
	- rollout_importance_sampling_threshold: Upper threshold for IS weights (default: 2.0)

### 2. Refactor GRPO Padding-Free Mode
Refactored _get_per_token_logps_and_entropies to:
- Compute logps on rmpad (remove-padding) tensors first
- Pad back to batch shape after computation using pad_logps_back_to_batch
- This makes loss_func padding-free-agnostic, simplifying downstream logic

### 3. Add Comprehensive Rollout Correction Metrics

Added off-policy diagnostic metrics (always logged when rollout_log_probs available):
- KL Divergence: kl (direct estimator), k3_kl (K3 estimator for stability)
- Perplexity: training_ppl, rollout_ppl, log_ppl_diff, ppl_ratio
- χ² Divergence: chi2_token, chi2_seq
- Added IS correction metrics (logged when IS mode enabled):
- is_weight_mean: Mean of importance sampling weights
- ess: Effective Sample Size (measures weight uniformity, range (0, 1])
- clipped_frac: Fraction of clipped/masked samples

All metrics are prefixed with rollout_correction/ in logs.

resolve https://github.com/modelscope/ms-swift/issues/6235

### CODE REVIEW

File: swift/trainers/rlhf_trainer/grpo_trainer.py [hunk 1]
```diff
@@ -828,15 +817,75 @@ def _prepare_batch_inputs(self, inputs: DataType) -> List[DataType]:
...
+                            # Standard mode: simple padding
+                            max_len = logits_to_keep
+                            padded_logprobs = []
+                            for lp in vllm_logprobs_list:
+                                # Take last logits_to_keep tokens
+                                lp_tensor = lp[-logits_to_keep:] if len(lp) >= logits_to_keep else lp
+                                # Pad if needed
+                                if len(lp_tensor) < max_len:
+                                    # right padding
+                                    lp_tensor = lp_tensor + [-1e10] * (max_len - len(lp_tensor))
```
[the remark this answers is not in the document]
Comment (hjh0119) [reply swift/trainers/rlhf_trainer/grpo_trainer.py:874]: In padding-free mode, left-pad to fill in the missing prompt part.
In non-padding-free mode, right-pad to match the pad portion in the corresponding encoding.
